cloudant-client/src/test/java/com/cloudant/api/query/IndexListTests.java

```diff
@@ -0,0 +1,249 @@
+            "\"fields\":[{\"Movie_name\":\"string\"}]," +
+            "\"index_array_lengths\":true}}";
+    private static String TEXT_COMPLEX = "{\"ddoc\":\"_design/testindexddoc\"," +
+            "\"name\":\"complextext\",\"type\":\"text\",\"def\":{\"default_analyzer\": " +
+            "{\"name\":\"perfield\",\"default\":\"english\",\"fields\":{\"spanish\":\"spanish\"," +
+            "\"german\":\"german\"}},\"default_field\":{\"enabled\": true, \"analyzer\": " +
+            "\"spanish\"},\"partial_filter_selector\":" + SELECTOR_STRING + "," +
+            "\"fields\":[{\"Movie_name\":\"string\"}, {\"Movie_runtime\": \"number\"}, " +
+            "{\"Movie_wonaward\": \"boolean\"}],\"index_array_lengths\":true}}";
+    private static String TEXT_ALL_FIELDS = "{\"ddoc\":\"_design/testindexddoc\"," +
+            "\"name\":\"textallfields\",\"type\":\"text\"," +
+            "\"def\":{\"default_analyzer\":\"keyword\",\"default_field\":{\"enabled\": false}," +
+            "\"partial_filter_selector\":{},\"fields\":[],\"index_array_lengths\":false}}";
+
+
+    private void enqueueList(String... indexes) {
+        MockResponse response = new MockResponse();
+        StringBuilder responseContent = new StringBuilder("{\"indexes\": [");
+        // Always add the special index
+        responseContent.append(ALL_DOCS);
+        // Add the rest of the indexes
+        for (String index : indexes) {
+            responseContent.append(",");
+            responseContent.append(index);
+        }
+        responseContent.append("], \"total_rows\": " + (indexes.length + 1) + "}");
+        response.setBody(responseContent.toString());
+        server.enqueue(response);
+    }
+
+    private void assertIndex(Index index, String name, String type, String selector) throws
+            Exception {
+        if (selector == null) {
+            selector = "{}";
+        }
+        assertIndex(index, name, "_design/testindexddoc", type, selector);
+    }
+
+    private void assertIndex(Index index, String name, String ddoc, String type, String selector)
+            throws Exception {
+        assertEquals("The index should have the correct name", name, index.getName());
+        assertEquals("The index should have the correct ddoc", ddoc, index
+                .getDesignDocumentID());
+        assertEquals("The index should have the correct type", type, index.getType());
+        assertEquals("The index should have the correct selector", selector, index
+                .getPartialFilterSelector());
+    }
+
+    private void assertJsonIndex(JsonIndex index, String name, String selector, JsonIndex
+            .Field... fields) throws Exception {
+        assertIndex(index, name, "json", selector);
+        // Assert the fields
+        assertEquals("The fields should be correct", Arrays.asList(fields), index.getFields());
+    }
+
+    private void assertTextIndex(TextIndex index, String name, String selector, String analyzer,
```

Comment:
Looking at this test helper makes me realise I hadn't thought about how/if we deserialise selectors. Obviously we have the builder stuff on my PR to help people build them if they want to use that to help them. What do we do about deserialisation?

I can see your approach is to keep it as a String, which is probably the most sensible, and if users want to use a JSON mapper etc to modify it they can. Or we could deserialise as a Map<String,Object> or whatever which is about as descriptive as we can get.

Reply:
The rationale for String was that it could be passed straight back in again (or into a new index builder) without any further manipulation.
I hope that we will be able to tweak your query builder slightly to allow output of a selector alone (since it's not a full query) to help with the building up. I don't think we need to deserialize into a model especially given that it wouldn't be supported by the proposed query builder (for editing as it were).